tests/cpgrid/addLgrsOnDistributedGrid_test.cpp

```diff
@@ -677,20 +676,66 @@ BOOST_AUTO_TEST_CASE(throw_not_fully_interior_lgr)
     }
 }
 
-//Calling globalRefine on a distributed grid is not supported yet.
+//Calling globalRefine on a distributed grid is supported now.
+BOOST_AUTO_TEST_CASE(globalRefine1)
+{
+    // Create a grid
+    Dune::CpGrid grid;
+    const std::array<double, 3> cell_sizes = {1.0, 1.0, 1.0};
+    const std::array<int, 3> grid_dim = {4,2,1};
+    grid.createCartesian(grid_dim, cell_sizes);
+
+    std::vector<int> parts(8);
+    std::vector<std::vector<int>> cells_per_rank = {{0,4},{1,5},{2,6}, {3,7}};
+    for (int rank = 0; rank < 4; ++rank) {
+        for (const auto& elemIdx : cells_per_rank[rank]) {
+            parts[elemIdx] = rank;
+        }
+    }
+    // Distribute the grid
+    if(grid.comm().size()>1)
+    {
+        grid.loadBalance();
+
+        grid.globalRefine(1);
```

Comment:
We should test other arguments, too.

Reply:
I added a call with argument equal 2, that - for now - throws.